apps/analytics/src/hooks/useReserve.tsx

```diff
@@ -1,33 +1,78 @@
 import { PrizePool } from '@generationsoftware/hyperstructure-client-js'
+import { useLastAwardedDrawId } from '@generationsoftware/hyperstructure-react-hooks'
 import { NO_REFETCH } from '@shared/generic-react-hooks'
 import { getSimpleMulticallResults, prizePoolABI } from '@shared/utilities'
 import { useQuery } from '@tanstack/react-query'
 import { usePublicClient } from 'wagmi'
+import { useDrawStatus } from './useDrawStatus'
 
 export const useReserve = (prizePool: PrizePool, options?: { refetchInterval?: number }) => {
   const publicClient = usePublicClient({ chainId: prizePool?.chainId })
 
-  const queryKey = ['reserve', prizePool?.chainId]
+  const { data: lastAwardedDrawId } = useLastAwardedDrawId(prizePool, {
+    refetchInterval: options?.refetchInterval
+  })
+
+  const { status: lastDrawStatus } = useDrawStatus(prizePool, lastAwardedDrawId as number)
+
+  const queryKey = ['reserve', prizePool?.chainId, lastAwardedDrawId, lastDrawStatus]
 
   return useQuery(
     queryKey,
     async () => {
-      const multicallResults = await getSimpleMulticallResults(
-        publicClient,
-        prizePool.address,
-        prizePoolABI,
-        [{ functionName: 'reserve' }, { functionName: 'pendingReserveContributions' }]
-      )
+      if (!!lastAwardedDrawId && !!lastDrawStatus) {
+        const multicallResults = await getSimpleMulticallResults(
+          publicClient,
+          prizePool.address,
+          prizePoolABI,
+          [
+            { functionName: 'reserve' },
+            { functionName: 'pendingReserveContributions' },
+            {
+              functionName: 'getTotalContributedBetween',
+              args: [lastAwardedDrawId + 1, lastAwardedDrawId + 1]
+            },
+            { functionName: 'numberOfTiers' }
+          ]
+        )
+
+        const current = typeof multicallResults[0] === 'bigint' ? multicallResults[0] : 0n
 
-      const current = typeof multicallResults[0] === 'bigint' ? multicallResults[0] : 0n
-      const pending = typeof multicallResults[1] === 'bigint' ? multicallResults[1] : 0n
+        const pending = typeof multicallResults[1] === 'bigint' ? multicallResults[1] : 0n
 
-      return { current, pending }
+        const contributions = typeof multicallResults[2] === 'bigint' ? multicallResults[2] : 0n
+        const numTiers = typeof multicallResults[3] === 'number' ? multicallResults[3] : 0
+        const pendingFallback = calculateFallbackPendingReserve(prizePool, contributions, numTiers)
+
+        if (lastDrawStatus === 'awarded' && !!pendingFallback) {
```

Comment:
`pendingFallback` can be zero at the beginning of a draw and still be a more accurate estimate than the value read from the contract. Might be good to get rid of the `!!pendingFallback` conditional and show zero if it's zero. Otherwise it can lead to an odd scenario where the inflated contract value is used at the very beginning of a draw.